File: packages/pluto/src/clients/shared/website.ts
import { utils } from "@plutolang/base";
import { IWebsiteClient, Website, WebsiteOptions } from "../../website";

export class WebsiteClient implements IWebsiteClient {
  private readonly id: string;

  constructor(path: string, name?: string, opts?: WebsiteOptions) {
    name = name ?? "default";
    this.id = utils.genResourceId(Website.fqn, name);
    path;
    opts;
  }

  public url(): string {
    return utils.getEnvValForProperty(this.id, "url");
  }
}

File: packages/pluto-py/pluto_client/clients/shared/website.py
from typing import Optional
from pluto_base.utils import gen_resource_id, get_env_val_for_property
from ...website import IWebsiteClient, Website, WebsiteOptions


class WebsiteClient(IWebsiteClient):
    def __init__(
        self,
        path: str,
        name: Optional[str] = None,
        opts: Optional[WebsiteOptions] = None,
    ):
        name = name or "default"
        self.__id = gen_resource_id(Website.fqn, name)

    def url(self) -> str:
        return get_env_val_for_property(self.__id, "url")
